src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarTownEventListener.java

```diff
@@ -107,13 +110,35 @@ public void onTownToggleExplosion(TownToggleExplosionEvent event) {
 	 */
 	@EventHandler
 	public void onTownTogglePVP(TownTogglePVPEvent event) {
-		if(SiegeWarSettings.getWarSiegeEnabled()
-				&& SiegeWarSettings.getWarSiegePvpAlwaysOnInBesiegedTowns()
-				&& SiegeController.hasActiveSiege(event.getTown()))  {
-			event.setCancellationMsg(Translation.of("plugin_prefix") + Translation.of("msg_err_siege_besieged_town_cannot_toggle_pvp"));
-			event.setCancelled(true);
+		if (SiegeWarSettings.getWarSiegeEnabled()) {
+			if (SiegeWarSettings.getWarSiegePvpAlwaysOnInBesiegedTowns() && SiegeController.hasActiveSiege(event.getTown()))  {
+				event.setCancellationMsg(Translation.of("plugin_prefix") + Translation.of("msg_err_siege_besieged_town_cannot_toggle_pvp"));
+				event.setCancelled(true);
+			}
+			if (SiegeWarSettings.getWarCommonPeacefulTownsEnabled() && event.getTown().isNeutral() && !event.getTown().isPVP()) {
+				event.setCancellationMsg(Translation.of("plugin_prefix") + Translation.of("msg_err_peaceful_town_pvp_forced_off"));
+				event.setCancelled(true);
+			}
 		}
 	}
+
+	@EventHandler
+	public void onPlotTogglePVP(PlotTogglePvpEvent event) {
+		if (SiegeWarSettings.getWarSiegeEnabled()) {
+			if (SiegeWarSettings.getWarSiegePvpAlwaysOnInBesiegedTowns() && SiegeController.hasActiveSiege(event.getTown()))  {
+				Messaging.sendErrorMsg(event.getPlayer(), Translation.of("plugin_prefix") + Translation.of("msg_err_siege_besieged_town_cannot_toggle_pvp"));
+				event.setCancelled(true);
+			}
+			if (SiegeWarSettings.getWarCommonPeacefulTownsEnabled() && event.getTown().isNeutral()) {
+				if (!event.getTown().isAdminDisabledPVP()) {
+					event.getTown().setAdminDisabledPVP(true);
+					TownyUniverse.getInstance().getDataSource().saveTown(event.getTown());

```

Comment:
Change to `event.getTown().save();`

Reply:
`event.getTown().save();` doesn't seem to be a valid function

Reply:
My bad again, that will be available once the pom and required versions are bumped. I'll resolve this one.